### PR TITLE
[Module] Added Microsoft.App/jobs

### PR DESCRIPTION
**Pull request by MrRoundRobin**
# Description

Add Module for Container App Jobs

### Pipeline references

| Pipeline |
| - |
| [App - Jobs](https://github.com/MrRoundRobin/AzureResourceModules/actions/workflows/ms.app.jobs.yml) |

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/app/job/.test/common/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+metadata name = 'Using large parameter set'
+metadata description = 'This instance deploys the module with most of its features enabled.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-app.containerApps-${serviceShort}-rg'
```
Comment (AlexanderSehr) [inline modules/app/job/.test/common/main.test.bicep:12]:
```suggestion
param resourceGroupName string = 'dep-${namePrefix}-app.jobs-${serviceShort}-rg'
```

File: modules/app/job/.test/common/main.test.bicep [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'mcajcom'
```
Comment (AlexanderSehr) [inline modules/app/job/.test/common/main.test.bicep:18]:
```suggestion
param serviceShort string = 'ajcom'
```

File: modules/app/job/.test/common/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    }
+    enableDefaultTelemetry: enableDefaultTelemetry
+    environmentId: nestedDependencies.outputs.managedEnvironmentResourceId
+    workloadProfileName: serviceShort
+    location: location
+    lock: {
+      kind: 'CanNotDelete'
+      name: 'myCustomLockName'
+    }
+    userAssignedIdentities: {
```
Comment (AlexanderSehr) [inline modules/app/job/.test/common/main.test.bicep:69]: This should already be implemented as per the latest specs

File: modules/app/job/.test/min/main.test.bicep [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+metadata name = 'Using only defaults'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-app.containerApps-${serviceShort}-rg'
```
Comment (AlexanderSehr) [inline modules/app/job/.test/min/main.test.bicep:12]:
```suggestion
param resourceGroupName string = 'dep-${namePrefix}-app.job-${serviceShort}-rg'
```

File: modules/app/job/.test/min/main.test.bicep [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'mcajmin'
```
Comment (AlexanderSehr) [inline modules/app/job/.test/min/main.test.bicep:18]:
```suggestion
param serviceShort string = 'ajmin'
```

File: modules/app/job/.test/common/main.test.bicep [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    managedIdentityName: 'dep-${namePrefix}-msi-${serviceShort}'
+    workloadProfileName: serviceShort
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../main.bicep' = {
```
Comment (AlexanderSehr) [inline modules/app/job/.test/common/main.test.bicep:52]: RBAC test is missing

File: modules/app/job/.test/common/main.test.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply modules/app/job/.test/common/main.test.bicep:12]: Requires re-generating the docs

**Comment by MrRoundRobin** [comment]
@AlexanderSehr addressed your feedback

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**